app/src/main/java/com/example/polina/socialnetwork/ProfileActivity.java

```diff
@@ -153,19 +181,30 @@ public void addProfileInfo(JSONObject o) {
 
     @UiThread
     public void updateAdapter(ArrayList<Post> posts) {
-        postList.setAdapter(adapter);
-        adapter.clear();
         adapter.addAll(posts);
         adapter.notifyDataSetChanged();
+        postsToLoad.addAll(posts);
```

Comment:
why do you add the same data twice??
just do:
postsToLoad.addAll(posts);
adapter.notifyDataSetChanged();